### ISSUE TITLE
Add unused constant/variables in ModelBasicCNN will lower the accuracy of black-box attack

### ISSUE THREAD

**Issue by chihchiehchen** [tensorflow/cleverhans#918]
Hello,

Thanks for implementing adversarial example library. When I try to modify the vanilla neural network ( ModelBasicCNN in cleverhans_tutorials/tutorial_models.py ), something strange happened: I put some 
unused variables on fprop(e.g. add mean=tf.zeros([10,1024]) in line 36 or 38), and run mnist_blackbox.py
(which was moved to the main folder), the prediction accuracy will not change (around 99.3%) but the test accuracy on adversarial examples will drop (from 70% to 54% ~65%).

I am not sure if I made something wrong. If anyone realizes what is going on,  that will really help me a lot.

In any case thanks a lot.

Best,

**Comment by npapernot** [comment]
Could you elaborate a bit on what the unused variables were? It is possible that if there is a greater mismatch between the substitute and the blackbox models, the transferability rate would decrease.

**Comment by chihchiehchen** [comment]
Dear Sir,

Yes. I simply use the ModelBasicCNN, but add one line (e.g. add mean=tf.zeros([10,1024] ) in line 36 or line 38, which was never used in the model). While it does not affect accuracy of clean data, I found the adversarial test accuracy changes.

I also tried some unused random initialization variable (e.g. mean=tf.Variable(tf.random_normal(shape, mean=0.0, stddev=6.0, dtype=tf.float32)) put on the same place) and similar things happened; in the beginning I tried to modify the network architecture to improve the performance, but looks like even I 
put some unused variable, it will still affect the adversarial detection.

It is also possible I really did something wrong (e.g. did not set up the environment well), in any case thanks for your paying attention.

Best regards,
Chih-Chieh

**Comment by npapernot** [comment]
Can you send a script that reproduces your issue?

**Comment by chihchiehchen** [comment]
Dear Sir,

Yes, here it is. 

git-hash : 510e16df371f006ded96ad7fbae7486653897aba

Modified code
```
--- a/cleverhans_tutorials/tutorial_models.py
+++ b/cleverhans_tutorials/tutorial_models.py
@@ -35,6 +35,7 @@ class ModelBasicCNN(Model):
         tf.layers.conv2d, activation=tf.nn.relu,
         kernel_initializer=initializers.HeReLuNormalInitializer)
     with tf.variable_scope(self.scope, reuse=tf.AUTO_REUSE):
+      mean=tf.zeros([10,1024])
       y = my_conv(x, self.nb_filters, 8, strides=2, padding='same')
       y = my_conv(y, 2 * self.nb_filters, 6, strides=2, padding='valid')
       y = my_conv(y, 2 * self.nb_filters, 5, strides=1, padding='valid')
ecchen@ecchen-B150M-D3H:/media/ecchen/dl/cleverhans$ git log -1
```

test script:
```
# workdir ${HOME}/cleverhans/
cp cleverhans_tutorials/mnist_blackbox.py   .
python mnist_blackbox.py

before modification:
Test accuracy of oracle on adversarial examples generated using the substitute: 0.702436548223502

after modification:
Test accuracy of oracle on adversarial examples generated using the substitute: 0.5667005076142132

I used Conda 4.5.11 , tensorflow 1.11.0 , Cuda 10.0, Python 3.6.7 

This is not a clean environment and so it is possible to be affected by other software, but I tried on the other computer and similar thing happened,  in any case thanks for your help.

Best Regards,
Chih-Chieh

**Comment by npapernot** [comment]
Hi Chih-Chieh, I cloned the head of our repository's master branch and added the diff described in your message but I could not reproduce your bug unfortunately (I still get the same accuracy of ~0.70). If you are able to isolate the bug, feel free to reopen this issue.
